### ISSUE TITLE
Pi didn't restart

### ISSUE THREAD

**Issue by tomsimmons78** [rcdrones/UPSPACK_V3#9]
I tested the V3 board last night, after 12 hours the STA signal caused the Pi to shutdown correctly.

At this point the power was about 5% I think, the last light was flashing, the power light was still on.

After the shutdown I plugged the mains supply to the UPS back in, the red light started flashing, power light was still on and over time time the battery level lights starts to climb.

It was my understanding that once there was a certainly amount of charge in the battery, Pi would be restarted, however this didn't happen.  However, the more I think about it, I believe the Pi power light was still lit.  If restarting the Pi is achieved simply by power being re-applied, does that mean that I would need to wait longer, until the UPS actually shuts down from lack of power?

If so, then that's a bit annoying, with nothing drawing power apart from the UPS, it's going to take a long time for that last 5% to be consumed, meaning that the only way to resume on power resumption would be via the power switch.

Tom

**Comment by tomsimmons78** [comment]
So as I expected, after allowing the UPS to shutdown the Pi (as it had before), and leaving it until the UPS lights were all off, when I re-applied the power to the UPS it started charging and after a short while the Pi started up.

It would be preferable if as soon as the power is re-established that if there is sufficient power in the battery the output is enabled.


Tom

**Comment by organizedkaos** [comment]
I agree.  If we can configure it that might help with managing the power for the end user.

For me, if I can get battery voltage (at the minimum) from the uart data then I can actually script a capacity monitor process that could take into consideration voltage drop and allow me to choose to shutdown the pi or otherwise.

**Comment by tomsimmons78** [comment]
@organizedkaos 

I'm not sure you can monitor the voltage, I believe it is using a buck converter so as the supply voltage drops it consumes more power to maintain the voltage.

When I came back to it at 12 hours it was showing the same voltage as the start.


Tom

**Comment by organizedkaos** [comment]
I mean the source voltage.  Yes, the boost convertor (or step-up) brings the voltage up to 5.xxx and maintains it but the source voltage (the 1s lipo) has a range of 4.2v (full) down to 3.4 (safe with a load).

When I tested the STA line the UPS didn't pulse until the capacity was at or below 5%.  Based on the voltage I was monitoring (at the source) this didn't happen until the lipo saw 3.2v with load (too low IMO).

If this UPS can't report source voltage then I would have to resort to using another hardware item to send the voltage range to the pi so that I can do the logic in python.

**Comment by new2thegame89** [comment]
I'm having the same issue. Even when the UPS is charged to 100% (all 4 LEDs ON), reapplying power to the USB-C port does not force the pi to restart. The only way it seems to work is if you manually toggle the power switch on the UPS. This is not super useful. Anyone been able to get this listed feature to work properly?

**Comment by tomsimmons78** [comment]
Once the Pi has been shutdown, you have to wait until the UPS turns itself off, then when power is restored and UPS is happy there is enough juice in the battery I think it will start the Pi.

Or that's what seemed to be the case, which isn't ideal, not least if the power came back just after the Pi was powered off.

I think this is the biggest weakness of this UPS, wish they would address, or even make the firmware source available.


Tom

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by new2thegame89** [comment]
Agree this is a big weakness. It also isn't clear from their description as they have it listed as the pi turning back on. Guess I'll need to look for another solution is the interim. Do you have any recommendations that you are familiar with? Thanks.

**Comment by zify85** [comment]
I realize this is a very old thread but I have the newer V3 version with the 3.2 firmware and I have the same issue.  If I remove source power, shutdown the pi (via python), and reapply power quickly the pi never comes back on.  It's almost like the recharging logic to turn the pi back on isnt working.  It does power the pi if I turn off/on the UPS switch which tells me its some sort of logic error in the fw.  

Any solutions after 4 years of experience?
